php/class-post-type.php

```diff
@@ -549,4 +548,37 @@ public function filter_user_has_cap( $allcaps, $caps ) {
 
 		return $allcaps;
 	}
+
+	/**
+	 * Publish snapshot changes when snapshot post is being published.
+	 *
+	 * @param int      $post_id Post id.
+	 * @param \WP_Post $post Post object.
+	 */
+	public function publish_snapshot( $post_id, $post ) {
+		if ( ! is_a( $this->snapshot_manager->customize_manager, '\WP_Customize_Manager' ) ) {
+			require_once( ABSPATH . WPINC . '/class-wp-customize-manager.php' );
+			$this->snapshot_manager->customize_manager = new \WP_Customize_Manager();
+		}
+		$snapshot_content = json_decode( $post->post_content, true );
+		if ( json_last_error() || ! is_array( $snapshot_content ) ) {
+			return;
+		}
+		$snapshot_values = array_filter(
+			wp_list_pluck( $snapshot_content, 'value' ),
+			function( $value ) {
+				return ! is_null( $value );
+			}
+		);
+
+		/** This action is documented in wp-includes/class-wp-customize-manager.php */
+		do_action( 'customize_save', $this->snapshot_manager->customize_manager );
+
+		foreach ( $snapshot_values as $setting_id => $setting_params ) {
+			$this->snapshot_manager->customize_manager->set_post_value( $setting_id, $setting_params );
+			$this->snapshot_manager->customize_manager->add_setting( $setting_id, $setting_params );
```

Comment:
@PatelUtkarsh this isn't going to work as expected. Instead, you should above do:

```
$this->snapshot_manager->customize_manager->add_dynamic_settings( array_keys( $snapshot_values ) );
```

This will ensure that the dynamic settings are added. For static settings, it will be important to do:

```php
do_action( 'customize_register', $this->snapshot_manager->customize_manager );
```